### ISSUE TITLE
CSS not copied

### ISSUE THREAD

**Issue by bernd-wechner** [zenorocha/clipboard.js#775]
## 🐛 Bug Report

> Fork this [JSFiddle](https://jsfiddle.net/zenorocha/5kk0eysw/) and reproduce your issue.

Demonstrated here: https://jsfiddle.net/thumbone/zy3daw0c/16/

That puts on my clipboard different things depending on whether I use Firefox of Chromeium. The latter brings more of the styling with it, neither of them brings colour from CSS vars.

### Have you read the [Contributing Guidelines on issues](https://github.com/zenorocha/clipboard.js/blob/master/contributing.md)?

Sure have. Thanks. It's brief enough to actually read not just skim to bottom and click "I've read it" like on most pop up licensing forms ;-)

### Expected Behaviour

I'd really like it to copy the CSS over as well. Or convert it all to inline before copying. That may be non trivial. I see two possible approaches and am too naive on the current industry options and standards in this space (which is why I lean on top notch offerings like yours ;-):

1. Find and include all the used CSS declarations in the copy, as incline CSS.
2. Convert all the CSS appplied styles to style attributes before copying

### Actual Behaviour

Mixed, depends on browser in part. But basically there's very little styling conserved in the copy.

### To Reproduce

The fiddle above suffices to demonstrate.

### Browsers Affected

I tested on Firefox, Chrome and Chromium.

### Operating System

I'm on Linux, Mint. 20.2.

**Comment by bernd-wechner** [comment]
OK, I sunk a whole day into exploring this and running tests. I stopped short of all my planned experiments because I stumbled near end of day upon the one solution that works! And it means I don't need `clipboard.js`. 

In summary, I looked at including the styles in the copy. I copy all the rules form the DOM into a new style element and insert that along with a clone of the element I wish to copy into a div wrapper. Then I copy that wrapper. And I tried so many methods and all of them trashed the style element. Firefox completely as a rule and losing all styles, and Chrome applies the styles to each element as element style attributes. Alas it applies the `var()`  calls literally and because the style element is dropped and it's not smart enough they get lost. 

Then late in the day (well in fact it's late at night now) I stumbled upon this and I pay homage to gods of chance and SO gems:

https://stackoverflow.com/a/45352464/4002633

I tuned that to copy the HTML (where wrapper is the DIV element into which I put a style element built from rules copied from the DOM followed by the content I want copied):

```lang-js
const HTML = wrapper.html();

function handler(event) {
    event.clipboardData.setData('text/html', HTML);
    event.preventDefault();
    document.removeEventListener('copy', handler, true);
}

document.addEventListener('copy', handler, true);
document.execCommand('copy');
```
And this copies perfectly truly. I can paste into an email (Thunderbird) and I get all my style,s I even get tooltips, animations and links ... it's complete! Mind blown. And I am not one to reach for that tacky idiom, as good as never, but after a hard day of research and experiments with two browsers, back and forth to have found a perfect solution is just stunning.

it could only be better if I could find documentation sources for this, to understand it and not just rest on someones claim and my empirical test that it does work. This line `event.clipboardData.setData('text/html', HTML);` alone keeps style elements in place, while and copy performed one a selection tosses them (they are I presume not selectable ;-).
